logstash-core/src/main/java/org/logstash/ext/JrubyEventExtLibrary.java

```diff
@@ -42,6 +42,8 @@
          */
         private final int hash = nextHash();
 
+        private final long sequence = SEQUENCE_GENERATOR.get();
```

Comment:
This feels a little fragile and potentially vulnerable to race conditions, since we are relying on a separate operation to increment the sequence generator and cannot guarantee that we will receive a globally-unique sequence ID. While this _may_ be moot in that an individual stream of events will at least be ever-increasing, providing globally-unique sequence ids is possible with:

~~~ diff
diff --git a/logstash-core/src/main/java/org/logstash/ext/JrubyEventExtLibrary.java b/logstash-core/src/main/java/org/logstash/ext/JrubyEventExtLibrary.java
index 369ea3fd0..e72e82844 100644
--- a/logstash-core/src/main/java/org/logstash/ext/JrubyEventExtLibrary.java
+++ b/logstash-core/src/main/java/org/logstash/ext/JrubyEventExtLibrary.java
@@ -36,16 +36,25 @@ public final class JrubyEventExtLibrary {
          */
         private static final AtomicLong SEQUENCE_GENERATOR = new AtomicLong(1L);
 
+        /**
+         * Sequence identifier for this instance. Used to make ordering guarantees.
+         */
+        private final long sequence;
+
         /**
          * Hashcode of this instance. Used to avoid the more expensive {@link RubyObject#hashCode()}
          * since we only care about reference equality for this class anyway.
          */
-        private final int hash = nextHash();
+        private final int hash;
 
         private Event event;
 
         public RubyEvent(final Ruby runtime, final RubyClass klass) {
             super(runtime, klass);
+
+            // generated fixed sequence id and hash code
+            sequence = SEQUENCE_GENERATOR.getAndIncrement();
+            hash = (int) (sequence ^ sequence >>> 32) + 31;
         }
 
         public static RubyEvent newRubyEvent(Ruby runtime) {
@@ -275,6 +284,10 @@ public final class JrubyEventExtLibrary {
             return value;
         }
 
+        public long sequence() {
+            return sequence;
+        }
+
         @Override
         public int hashCode() {
             return hash;
@@ -342,14 +355,5 @@ public final class JrubyEventExtLibrary {
         private void setEvent(Event event) {
             this.event = event;
         }
-
-        /**
-         * Generates a fixed hashcode.
-         * @return HashCode value
-         */
-        private static int nextHash() {
-            final long sequence = SEQUENCE_GENERATOR.incrementAndGet();
-            return (int) (sequence ^ sequence >>> 32) + 31;
-        }
     }
 }
~~~